File: sdks/flutter/ios/Classes/SibnaFlutterPlugin.swift
import Flutter
import UIKit

/// SibnaFlutterPlugin — iOS host-side plugin.
///
/// The Sibna Flutter plugin uses dart:ffi (ffiPlugin: true) for all native
/// calls. This class exists only to satisfy the Flutter plugin registry.
///
/// The native Rust library (libsibna.a) is statically linked via Xcode.
/// Build steps:
///   1. cargo build --release --target aarch64-apple-ios --features ffi
///   2. cargo build --release --target x86_64-apple-ios --features ffi
///   3. lipo -create target/aarch64-apple-ios/release/libsibna.a \
///            target/x86_64-apple-ios/release/libsibna.a \
///            -output ios/libsibna.a
public class SibnaFlutterPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // No MethodChannel needed — all calls go through dart:ffi
    }
}

File: sdks/flutter/android/src/main/kotlin/dev/sibna/flutter/SibnaFlutterPlugin.kt
package dev.sibna.flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * SibnaFlutterPlugin — Android host-side plugin.
 *
 * The Sibna Flutter plugin uses dart:ffi (ffiPlugin: true) for all
 * native calls. This class exists only to satisfy the Flutter plugin
 * registry — no MethodChannel is needed.
 *
 * The native Rust library (libsibna.so) is loaded automatically by
 * the Dart FFI layer via DynamicLibrary.open('libsibna.so').
 * It must be compiled for the target ABI and placed in:
 *   android/src/main/jniLibs/<ABI>/libsibna.so
 *
 * Supported ABIs: arm64-v8a, armeabi-v7a, x86_64
 */
class SibnaFlutterPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // No MethodChannel needed — all calls go through dart:ffi
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // Nothing to clean up
    }
}
